### PR TITLE
Fix: prevent possible int32 overflow in bmqt::UriParser

### PR DESCRIPTION
**Pull request by 678098**
https://github.com/bloomberg/blazingmq/issues/213

### CODE REVIEW

File: src/groups/bmq/bmqt/bmqt_uri.cpp [hunk 1]
```diff
@@ -193,6 +193,7 @@ void UriParser::initialize(bslma::Allocator* allocator)
     //       not work correctly with versions of IBM xlc12 released following
     //       the 'Dec 2015 PTF'.
     ++s_initialized;
+    BSLS_ASSERT(s_initialized > 0);
```
Comment (678098) [inline src/groups/bmq/bmqt/bmqt_uri.cpp:196]: This `assert` might be an overkill if we use int64 type, because it's way too difficult to cause an overflow of int64 variable here by incrementing by one.

We have another option to keep a counter as `int` like before or use `AtomicInt` and fail on this `assert` on 2.147kk calls.

File: src/groups/bmq/bmqt/bmqt_uri.t.cpp [hunk 2]
```diff
@@ -657,6 +657,28 @@ static void test7_testLongUri()
...
+    // Disable the default allocator check. When 'bmqt::Uri'
+    // is created from a long uri string error message is
+    // logged via 'BALL_LOG_ERROR' which allocates using
+    // the default allocator.
+
+    mwctst::TestHelper::printTestName("URI PARSER OVERFLOW TEST");
+
+    // Initialize bmqt::UriParser many times, and then deinitialize
+    // it the same number of times according to the usage contract.
+    const bsls::Types::Int64 k_INIT_NUM = 2200000000LL;
```
Comment (pniedzielski) [inline src/groups/bmq/bmqt/bmqt_uri.t.cpp]: Would `static_cast<int64_t>( numeric_limits<int32_t>::max() ) + 1` (or corresponding BDE-ism) be clearer?  It's a somewhat lower number of `initialize`/`shutdown`s, but says more clearly what we're trying to test.

File: src/groups/bmq/bmqt/bmqt_uri.t.cpp [hunk 3]
```diff
@@ -657,6 +657,28 @@ static void test7_testLongUri()
...
+    for (size_t i = 0; i < k_INIT_NUM; i++) {
```
Comment (pniedzielski) [inline src/groups/bmq/bmqt/bmqt_uri.t.cpp]: Does this give a signed/unsigned comparison warning?  I think picking either `size_t` or `Int64` is the right path here.

Comment (pniedzielski) [reply src/groups/bmq/bmqt/bmqt_uri.t.cpp]: Oh nvm a later commit did exactly this change.

File: src/groups/bmq/bmqt/bmqt_uri.cpp (hunk 1 above)
Comment (pniedzielski) [reply src/groups/bmq/bmqt/bmqt_uri.cpp:196]: Signed integer overflow is undefined behavior.  Propose moving this assert one line prior and changing it to `s_initialized < numeric_limits<int64_t>::max()` (or corresponding BDE-ism).

File: src/groups/bmq/bmqt/bmqt_uri.t.cpp [hunk 4]
```diff
@@ -657,28 +657,6 @@ static void test7_testLongUri()
     bmqt::UriParser::shutdown();
 }
 
-static void test8_testUriParserOverflow()
```
Comment (pniedzielski) [inline src/groups/bmq/bmqt/bmqt_uri.t.cpp]: Why is this removed again?  Does it take too long to run?  It seems like a useful test.

**Review by pniedzielski — CHANGES_REQUESTED** [review]

File: src/groups/bmq/bmqt/bmqt_uri.t.cpp (hunk 2 above)
Comment (678098) [reply src/groups/bmq/bmqt/bmqt_uri.t.cpp]: I removed this test from this branch, because it was too slow, 43 seconds on my M2.

File: src/groups/bmq/bmqt/bmqt_uri.t.cpp (hunk 4 above)
Comment (678098) [reply src/groups/bmq/bmqt/bmqt_uri.t.cpp]: 43 seconds, and overall this test doesn't test a useful behaviour and instead it tests if we have a bug or not.

**Review by pniedzielski — APPROVED** [review]
👍🏻
